controllers/network_controller.go

```diff
@@ -312,64 +326,114 @@ func (r *NetworkReconciler) reconcilePeeredVNIs(ctx context.Context, log logr.Lo
+				if err := r.MetalnetMBClient.CleanupNotPeeredRoutes(ctx, peeredVNI); err != nil {
+					errs = append(errs, err)
+					continue
 				}
 			} else {
-				if err := r.MetalnetMBClient.CleanupNotPeeredRoutes(vni); err != nil {
-					return err
+				if err := r.MetalnetMBClient.CleanupNotPeeredRoutes(ctx, vni); err != nil {
+					errs = append(errs, err)
+					continue
 				}
 			}
+
+			delete(statusPeerVnis, peeredVNI)
 		}
 
 		for _, peeredVNI := range added.UnsortedList() {
 			if !ownVniAvail {
 				return nil
 			}
+			statusPeerVnis[peeredVNI] = metalnetv1alpha1.NetworkPeeringStatePending
 			log.V(1).Info("Checking the existence of the peeredVNI in dp-service (added)", "peeredVNI", peeredVNI)
 			peeredVniAvail, err := r.DPDK.GetVni(ctx, peeredVNI, 0)
 			if err != nil {
-				return err
+				statusPeerVnis[peeredVNI] = metalnetv1alpha1.NetworkPeeringStateError
+				errs = append(errs, err)
+				continue
 			}
 			log.V(1).Info("Checked the existence of the peeredVNI in dp-service (added)", "peeredVNI", peeredVNI, "peeredVniAvail", peeredVniAvail)
 			if err := r.MetalnetCache.AddVniToPeerVnis(vni, peeredVNI); err != nil {
-				return err
+				statusPeerVnis[peeredVNI] = metalnetv1alpha1.NetworkPeeringStateError
+				errs = append(errs, err)
+				continue
 			}
 			if ownVniAvail && !peeredVniAvail.Spec.InUse {
 				if err := r.subscribeIfNotSubscribed(ctx, peeredVNI); err != nil {
-					return err
+					statusPeerVnis[peeredVNI] = metalnetv1alpha1.NetworkPeeringStateError
+					errs = append(errs, err)
+					continue
 				}
 			}
 			if ownVniAvail && peeredVniAvail.Spec.InUse {
 				if err := r.recycleVNISubscription(ctx, vni); err != nil {
-					return err
+					statusPeerVnis[peeredVNI] = metalnetv1alpha1.NetworkPeeringStateError
+					errs = append(errs, err)
+					continue
 				}
 				if err := r.recycleVNISubscription(ctx, peeredVNI); err != nil {
-					return err
+					statusPeerVnis[peeredVNI] = metalnetv1alpha1.NetworkPeeringStateError
+					errs = append(errs, err)
+					continue
 				}
 			}
+
+			statusPeerVnis[peeredVNI] = metalnetv1alpha1.NetworkPeeringStateReady
 		}
 	}
 
+	newStatusPeerings := make([]metalnetv1alpha1.NetworkPeeringStatus, 0, len(statusPeerVnis))
+	for peeredId, status := range statusPeerVnis {
+		newStatusPeerings = append(newStatusPeerings, metalnetv1alpha1.NetworkPeeringStatus{
+			ID:    int32(peeredId),
+			State: status,
+		})
+
+	}
+	log.V(1).Info("Updating network status peerings", "", newStatusPeerings)
+	if err := r.patchStatus(ctx, network, func() {
+		network.Status.Peerings = newStatusPeerings
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("error patching network peerings status: %w", err))
+	}
+
+	if err := errors.Join(errs...); err != nil {
+		return fmt.Errorf("error reconciling peered VNIs: %w", err)
+	}
+	return nil
+}
+
+func (r *NetworkReconciler) patchStatus(ctx context.Context, network *metalnetv1alpha1.Network, mutate func()) error {
+	base := network.DeepCopy()
+
+	mutate()
```

Comment:
What is the purpose of this `mutate()` function?

Reply:
It it just calling the function we are passing as a parameter to this PatchStatus method(Basically updating status)

` r.patchStatus(ctx, network, func() {
		network.Status.Peerings = newStatusPeerings
	});`
 Just to keep it consistent with other parts of the metalnet code, i have used patch method like, otherwise we can use our usual way of patching.